Yoast/wordpress-seo
# Add notification for terms

## Summary

This PR can be summarized in the following changelog entry:

* Show upsell notificifation when a term is removed.

## Relevant technical choices:

* When a term is deleted, check whether it is a publicly viewable term (i.e. whether its taxonomy is public or at least publicly queryable), and if so, add a notification about the deletion.
* Use the taxonomy singular name for the "You just deleted a {$taxname}." message. While in English this should preferably be a lower-cased word, we shouldn't assume anything here and just use the singular name as is.
* Load `wp-seo-quick-edit-handler.js` on both posts list screens and terms list screens.
* Hook into the `ajaxComplete` event in JS to detect whether a term has just been deleted. This is unfortunately the only way to recognize this event in JS. Then, fetch notifications from the server so that the notification about term deletion is immediately displayed.
* Run the existing logic for slug changes for both the posts and terms screens.
* Improve some of the existing logic, getting rid of `if` statements by using a variable to store the name of the field that contains the slug (`slug` for terms, `post_name` for posts).
* Load the `WPSEO_Slug_Change_Watcher` class outside of the admin class, as it must be watching for slug changes from anywhere. Not only the admin allows those to happen, it may also be the REST API, an AJAX request (like here), or anything else.
* While adding unit tests for the new functionality, I improved performance of the existing tests for that class by only creating as few posts as possible and instead reusing them.

## Test instructions

This PR can be tested by following these steps:

* Go to a terms list screen (for example "Categories" or "Tags").
* Add a term.
* Remove that term again.
* Ensure that right after removal of the term, you'll see a notification to highlight this action.

## Quality assurance


Fixes #
